src/Features/CSharp/Portable/CodeFixes/FixReturnType/CSharpFixReturnTypeCodeFixProvider.cs

```diff
@@ -0,0 +1,141 @@
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(
+               new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+               context.Diagnostics);
+        }
+
+        private async Task<(TypeSyntax declarationToFix, TypeSyntax fixedDeclaration)> TryGetOldAndNewReturnTypeAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            Debug.Assert(diagnostics.Length == 1);
+            var location = diagnostics[0].Location;
+            var returnStatement = (ReturnStatementSyntax)location.FindNode(getInnermostNodeForTie: true, cancellationToken: cancellationToken);
+
+            var returnedValue = returnStatement.Expression;
+            if (returnedValue == null)
+            {
+                return default;
+            }
+
+            var (declarationTypeToFix, useTask) = TryGetDeclarationTypeToFix(returnStatement);
+            if (declarationTypeToFix == null)
+            {
+                return default;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var returnedType = semanticModel.GetTypeInfo(returnedValue, cancellationToken).Type;
+            if (returnedType == null)
+            {
+                return default;
+            }
+
+            var fixedType = useTask
+                ? semanticModel.Compilation.TaskOfTType().Construct(returnedType)
+                : returnedType;
+            var fixedDeclaration = fixedType.GenerateTypeSyntax().WithTriviaFrom(declarationTypeToFix);
+
+            return (declarationTypeToFix, fixedDeclaration);
+        }
+
+        protected override async Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var (declarationTypeToFix, fixedDeclaration) =
+                await TryGetOldAndNewReturnTypeAsync(document, diagnostics, cancellationToken).ConfigureAwait(false);
+
+            editor.ReplaceNode(declarationTypeToFix, fixedDeclaration);
+            return;
+        }
+
+        private static (TypeSyntax type, bool useTask) TryGetDeclarationTypeToFix(SyntaxNode node)
+        {
+            if (!node.IsKind(SyntaxKind.ReturnStatement))
+            {
+                return default;
+            }
+
+            var containingMember = node.GetAncestors().FirstOrDefault(a => a.IsKind(
+                SyntaxKind.MethodDeclaration, SyntaxKind.PropertyDeclaration, SyntaxKind.ParenthesizedLambdaExpression, SyntaxKind.SimpleLambdaExpression,
+                SyntaxKind.LocalFunctionStatement, SyntaxKind.AnonymousMethodExpression, SyntaxKind.ConstructorDeclaration, SyntaxKind.DestructorDeclaration,
+                SyntaxKind.OperatorDeclaration, SyntaxKind.IndexerDeclaration, SyntaxKind.EventDeclaration));
```

Comment:
ick.  feels like there's something in the IDE somewhere for this :)

Reply:
Looked around, but didn't find anything good. Ended up removing it.